api/src/shared/domain/services/placement-profile-service.js

```diff
@@ -130,19 +130,25 @@ async function _generatePlacementProfile({ userId, profileDate, competences, all
   });
 }
 
-async function getPlacementProfilesWithSnapshotting({ userIdsAndDates, competences, allowExcessPixAndLevels = true }) {
-  const knowledgeElementsByUserIdAndDates =
-    await knowledgeElementSnapshotRepository.findMultipleUsersFromUserIdsAndSnappedAtDates(userIdsAndDates);
-
-  return userIdsAndDates.map(({ userId, sharedAt }) => {
-    const keForUser = knowledgeElementsByUserIdAndDates.find((knowledgeElementsByUserIdAndDates) => {
-      const sameUserId = knowledgeElementsByUserIdAndDates.userId === userId;
-      const sameDate = sharedAt && knowledgeElementsByUserIdAndDates.snappedAt.getTime() === sharedAt.getTime();
+async function getPlacementProfilesWithSnapshotting({
+  participationInfos,
```

Comment:
suggestion: participations
```suggestion
  participation,
```